### PR TITLE
clean up header queries

### PR DESCRIPTION
**Pull request by tjheffner**
## Description
Relates to #183. @ryguyk pointed out how to use addInclude properly, so we don't need to make additional queries for the promo blocks. They correctly come along for the ride with the menu data, which simplifies some of the logic when constructing the data into shape. Nice! 

## Testing done
- visit any story or story listing page, check the megamenu and footer and see that promo blocks are present on sections that have them in the cms (top-level `About VA`, or the `Disability`, `Education` hubs underneath the first item)

## QA steps
```[tasklist]
- [ ] Automated tests have passed
- [ ] Tugboat environment was generated without errors
```

## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:

### CODE REVIEW

File: src/data/queries/headerFooter.ts [hunk 1]
```diff
@@ -7,82 +7,68 @@ import {
...
+      'field_promo_reference.field_promo_link',
+    ])
 }
 
 // Define the option types for the data loader.
 type DataOpts = QueryOpts<{
   params?: object
   megaMenuParams?: object
-  promoBlockParams?: object
 }>
```
Comment (ryguyk) [inline src/data/queries/headerFooter.ts:40]: This should not be necessary. Our `data` function does not actually require any input `opts`. `params` and `megaMenuParams` are defined within this file and can be accessed directly as we now have it below. We should be able to remove this type definition (as well as its usage below; more commentary there).

File: src/data/queries/headerFooter.ts [hunk 2]
```diff
@@ -7,82 +7,68 @@ import {
...
 
 export const data: QueryData<DataOpts, RawHeaderFooterData> = async (opts) => {
```
Comment (ryguyk) [inline src/data/queries/headerFooter.ts:42]: This function does not actually require any input. I think we can change to:
```
export const data: QueryData<null, RawHeaderFooterData> = async () => {
```

File: src/lib/utils/header.ts [hunk 3]
```diff
@@ -105,41 +106,29 @@ function makeColumns(hostUrl, linkData, arrayDepth, promo): MegaMenuLinkObject {
     }
 
     // Generate promo block column if present
-    if (promo !== undefined || null) {
-      columns[columnNames[i]] = promo
+    if (promo && promo.id) {
+      columns[columnNames[i]] = queries.formatData(
+        'block_content--promo',
+        promo
+      )
```
Comment (ryguyk) [inline src/lib/utils/header.ts:113]: I find myself wondering if we still need a separate query file for block promos. Will we ever need to query this separately? Or should the formatting logic for promos be moved into this file?

**Review by ryguyk — CHANGES_REQUESTED** [review]
One change request and one question for discussion.

File: src/data/queries/headerFooter.ts (hunk 1 above)
Comment (tjheffner) [reply src/data/queries/headerFooter.ts:40]: good callout, will nuke

File: src/lib/utils/header.ts (hunk 3 above)
Comment (tjheffner) [reply src/lib/utils/header.ts:113]: I am not 100% sure yet. These promo blocks are also included on the main benefits hubs pages in addition to the header megamenu. I think the safe route is to keep the formatting logic separate from this header file for now. 

The query may be redundant but the formatting will be used in both places

Comment (ryguyk) [reply src/lib/utils/header.ts:113]: Sounds fair!

**Review by ryguyk — APPROVED** [review]

**Pull request merged**